### PR TITLE
Added "Add Folder" and "Add Divider" Button Functionality

### LINKED ISSUE #30: Backend and UI: Add Folder Button Functionality
- Add functionality to add folder button in the overlay age
- Add drop down for dividers in the overlay page for "Add Folder"
- Depends on Folder model in the backend #27

### LINKED ISSUE #33: Backend: Add Divider Functionality
- Add functionality to the add divider button in the overlay page
- Should push divider to the backend using API call 
- Purpose: should create a divider in the sidebar
- Depends on #32

### PR DESCRIPTION
**Pull request by ishaansathaye**
- Made backend requests to get the dividers for a user in the dropdown for dividers
- Made backend post request to send the new folder created that is now associated with a divider into the user
- Took care of an edge case where the fields are empty

### CODE REVIEW

**Review by kct417 — APPROVED** [review]
Looks good for now. As discussed in person, we may need to refactor the frontend such as Overlay, MenuPopup, and SideBar at a later time.

**Review by arnavbhola123 — APPROVED** [review]
Look great to me, good job with the code!

**Comment by ishaansathaye** [comment]
- Fixed some edge cases with the code
- Should work as intended now

**Review by ravippanchal — APPROVED** [review]
LGTM

**Comment by ishaansathaye** [comment]
- Finished adding divider functionality
- We are now able to add dividers and folders through the app for users (edge cases such as none entered are in place too)

File: packages/react-frontend/src/components/Sidebar.jsx [hunk 1]
```diff
@@ -90,14 +172,14 @@ const Sidebar = () => {
...
-						options: ['Divider 1', 'Divider 2', 'Divider 3'], // Add your divider options here
+						options: getDividersNames(),
 					},
 				];
 				buttons = [
 					{
 						label: 'Add Folder',
 						type: 'button',
-						onClick: () => alert('Folder added!'),
+						onClick: handleAddFolder,
```
Comment (charlescohanlon) [inline packages/react-frontend/src/components/Sidebar.jsx]: I think this should be an arrow function that is passed some kind of identifying info about the new folder

**Comment by ravippanchal** [comment]
LGTM

**Review by jlowe11 — APPROVED** [review]
Great job! LGTM!

**Review by charlescohanlon — APPROVED** [review]
if it works it works

**Pull request merged**